### PR TITLE
Uploader 2

### PR DESCRIPTION
**Pull request by john-hill**

### CODE REVIEW

File: src/main/java/org/sagebionetworks/web/client/widget/table/modal/UploadTableModalWidgetImpl.java [hunk 1]
```diff
@@ -47,22 +50,27 @@ public void configure(String parentId, TableCreatedHandler handler) {
...
-	 * Called when the file handle is created.
-	 * @param fileHandlId
-	 */
-	private void onFileHandleCreated(String fileHandlId){
-		
+
+	@Override
+	public void uploadSuccess(String fileHandleId) {
+		view.setPrimaryEnabled(true);
+		view.showError("Uploaded file handle: "+fileHandleId);
```
Comment (jay-hodgson) [inline src/main/java/org/sagebionetworks/web/client/widget/table/modal/UploadTableModalWidgetImpl.java]: suggest rename based on use (perhaps showAlert and setAlertVisible)

File: src/main/resources/org/sagebionetworks/web/Portal.gwt.xml [hunk 2]
```diff
@@ -59,9 +59,9 @@
 	<inherits name="org.sagebionetworks.gwt.JSONAdapter"/>
 	<inherits name="org.sagebionetworks.client.synpaseClientExceptions"/>
 	
-	<!-- Limit build permuatations (testing only)
+	<!-- Limit build permuatations (testing only) -->
 	<set-property name="user.agent" value="safari"/>
```
Comment (jay-hodgson) [inline src/main/resources/org/sagebionetworks/web/Portal.gwt.xml:63]: careful!  do not limit compile to safari only

File: src/main/java/org/sagebionetworks/web/client/widget/table/modal/UploadTableModalWidgetImpl.java [hunk 3]
```diff
@@ -47,22 +50,27 @@ public void configure(String parentId, TableCreatedHandler handler) {
...
+		view.setErrorVisible(true);
+	}
+
+	@Override
+	public void uploadFailed(String error) {
+		view.setPrimaryEnabled(true);
```
Comment (john-hill) [inline src/main/java/org/sagebionetworks/web/client/widget/table/modal/UploadTableModalWidgetImpl.java:71]: hide progress on error

File: src/main/resources/org/sagebionetworks/web/client/widget/table/modal/UploadTableModalViewImpl.ui.xml [hunk 4]
```diff
@@ -9,7 +9,9 @@
 	<b:Modal title="Upload Table" closable="true" fade="true"
 		dataBackdrop="STATIC" dataKeyboard="true">
 		<b:ModalBody>
+			<b:Alert ui:field="instructions" type="INFO" visible="false"></b:Alert>
```
Comment (john-hill) [inline src/main/resources/org/sagebionetworks/web/client/widget/table/modal/UploadTableModalViewImpl.ui.xml]: no alert for the instructions

File: src/main/resources/org/sagebionetworks/web/Portal.gwt.xml [hunk 5]
```diff
@@ -59,9 +59,9 @@
 	<inherits name="org.sagebionetworks.gwt.JSONAdapter"/>
 	<inherits name="org.sagebionetworks.client.synpaseClientExceptions"/>
 	
-	<!-- Limit build permuatations (testing only)
+	<!-- Limit build permuatations (testing only) -->
 	<set-property name="user.agent" value="safari"/>
-	  -->
```
Comment (john-hill) [inline src/main/resources/org/sagebionetworks/web/Portal.gwt.xml:64]: move back

File: src/main/java/org/sagebionetworks/web/client/widget/upload/FileInputWidgetImpl.java [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+				new ProgressingFileUploadHandler() {
+					@Override
+					public void uploadSuccess(String fileHandleId) {
+						// Set the view at 100%
+						view.updateProgress(100, "100%");
+						handler.uploadSuccess(fileHandleId);
+					}
+
+					@Override
+					public void uploadFailed(String error) {
```
Comment (john-hill) [inline src/main/java/org/sagebionetworks/web/client/widget/upload/FileInputWidgetImpl.java:52]: enable input on failure

**Pull request merged**